@here/harp-mapview/lib/Utils.ts

```diff
@@ -377,24 +404,90 @@ export namespace MapViewUtils {
             distance: unconstrained.distance
         };
 
+        if (projection.type === ProjectionType.Planar) {
+            if (scale > 1) {
+                constrained.distance /= scale;
+                camera
+                    .getWorldDirection(camera.position)
+                    .multiplyScalar(-constrained.distance)
+                    .add(worldMaxBounds.getCenter(constrained.target));
+            } else {
+                const targetBounds = cache.box3[0]
+                    .copy(worldMaxBounds as THREE.Box3)
+                    .expandByVector(viewHalfSize.multiplyScalar(-1));
+                targetBounds
+                    .clampPoint(unconstrained.target, constrained.target)
+                    .setZ(unconstrained.target.z);
+                if (constrained.target.equals(unconstrained.target)) {
+                    return unconstrained;
+                }
+
+                camera.position.x += constrained.target.x - unconstrained.target.x;
+                camera.position.y += constrained.target.y - unconstrained.target.y;
+            }
+            return constrained;
+        }
+
+        // Spherical projection
         if (scale > 1) {
+            // Set target to center of max bounds but keeping same height as unconstrained target.
```

Comment:
Isn't that height always the earth radius or do we allow the target to "fly"? Would make things easier to understand if you use `EarthConstants.EQUATORIAL_RADIUS` here

Reply:
I wanted to make it a little more flexible to support 3D terrain.

Reply:
okay, makes sense.